File: java/DesignPatterns/CreationalDesignPatterns/SingletonDesignPattern/SingletonDesignPattern.Example2/src/main/java/creational/singleton/example2/Singleton.java
package creational.singleton.example2;

// Eager Initialization: instance is created when the class is loaded,
// equivalent to C#'s static constructor approach.
public class Singleton {

    private static final Singleton instance = new Singleton();

    private Singleton() { }

    public static Singleton getInstance() {
        return instance;
    }
}

File: nodejs/DesignPatterns/CreationalDesignPatterns/SingletonDesignPattern/SingletonDesignPattern.Example2/src/Singleton.ts
// Eager initialization: instance is created when the class is first loaded, regardless of whether it is actually used or not.

export class Singleton {
  private static _idCounter: number = 0;
  public readonly id: number;

  private static readonly _instance: Singleton = new Singleton();

  private constructor() {
    Singleton._idCounter++;
    this.id = Singleton._idCounter;
  }

  static get getInstance(): Singleton {
    return Singleton._instance;
  }
}
